lms/static/js/learner_dashboard/views/entitlement_unenrollment_view.js

```diff
@@ -0,0 +1,149 @@
+
+                    this.$submitButton = $(this.submitButtonSelector);
+                    this.$trackInfo = $(this.trackInfoSelector);
+                    this.$refundInfo = $(this.refundInfoSelector);
+                    this.$errorInfo = $(this.errorInfoSelector);
+
+                    this.$submitButton.on('click', this.handleSubmit.bind(this));
+
+                    $(this.triggerSelector).each(function() {
+                        var $trigger = $(this);
+
+                        $trigger.on('click', view.handleTrigger.bind(view));
+
+                        if (window.accessible_modal) {
+                            window.accessible_modal(
+                                '#' + $trigger.attr('id'),
+                                view.closeButtonSelector,
+                                '#' + view.$el.attr('id'),
+                                view.mainPageSelector
+                            );
+                        }
+                    });
+                },
+
+                handleTrigger: function(event) {
+                    var $trigger = $(event.target),
+                        courseName = $trigger.data('courseName'),
+                        courseNumber = $trigger.data('courseNumber'),
+                        isRefundable = $trigger.data('entitlementIsRefundable'),
+                        apiEndpoint = $trigger.data('entitlementApiEndpoint');
+
+                    // Convert to boolean.
+                    isRefundable = isRefundable && isRefundable.toLowerCase() === 'true';
+
+                    this.resetModal();
+                    this.setTrackInfo(courseName, courseNumber);
+                    this.setRefundInfo(isRefundable);
+                    this.setSubmitData(apiEndpoint);
+
+                    if (window.edx && window.edx.dashboard && window.edx.dashboard.dropdown) {
+                        window.edx.dashboard.dropdown.toggleCourseActionsDropdownMenu(event);
+                        this.$el.css('position', 'fixed');
+                    }
+                },
+
+                handleSubmit: function() {
+                    var apiEndpoint = this.$submitButton.data('entitlementApiEndpoint');
+
+                    if (apiEndpoint === undefined) {
+                        this.setError(gettext('Error: cannot process unenrollment request.'));
+                        return;
+                    }
+
+                    $.ajax({
+                        url: apiEndpoint,
+                        method: 'DELETE',
+                        complete: this.onComplete.bind(this)
+                    });
+                },
+
+                resetModal: function() {
+                    this.$submitButton.removeData();
+                    this.$submitButton.prop('disabled', false);
+                    this.$trackInfo.empty();
+                    this.$refundInfo.empty();
+                    this.$errorInfo.css('display', 'none');
```

Comment:
We shouldn't be using JS to add inline styles. It's preferable to add CSS classes that contain the styles.

Reply:
changed.